apps/dashboard/src/components/workflow-editor/workflow-editor-provider.tsx

```diff
@@ -153,7 +162,27 @@ export const WorkflowEditorProvider = ({ children }: { children: ReactNode }) =>
     },
   });
 
-  const blocker = useBlocker(isDirty || isPending);
+  /*
+   * If there was a pending navigation when saving was in progress,
+   * proceed with that navigation now that changes are saved
+   *
+   * small timeout to briefly show the success dialog before navigating
+   */
+  const handleBlockedNavigation = useCallback(async () => {
+    return new Promise<void>((resolve) => {
+      setTimeout(() => {
+        blocker.proceed?.();
+        toast.dismiss();
+        resolve();
+      }, 500);
+    });
+  }, [blocker]);
```

Comment:
The async and timeout here I think its needed because it:
1. ensures users perceive the success state (better UX)
2. allows success animations to complete their transitions
3. prevents jarring immediate redirects that might make users question if the save completed